### DISCUSSION TITLE
Failed to connect to PRIME GRPC server

### DISCUSSION THREAD

**Discussion by ragulselvaraj20** [ansys/pyprimemesh#658] [Q&A] [answered]
I am getting the following error.

Failed to connect to PRIME GRPC server
Traceback (most recent call last):

  File ~\.conda\envs\myenv\Lib\site-packages\ansys\meshing\prime\internals\grpc_communicator.py:86 in __init__
    grpc.channel_ready_future(self._channel).result(timeout=timeout)

  File ~\.conda\envs\myenv\Lib\site-packages\grpc\_utilities.py:162 in result
    self._block(timeout)

  File ~\.conda\envs\myenv\Lib\site-packages\grpc\_utilities.py:106 in _block
    raise grpc.FutureTimeoutError()

FutureTimeoutError

The above exception was the direct cause of the following exception:

Traceback (most recent call last):

  File ~\.conda\envs\myenv\Lib\site-packages\spyder_kernels\py3compat.py:356 in compat_exec
    exec(code, globals, locals)

  File c:\projects\inverter\pyansys\examples\check1\untitled1.py:13
    prime_client = prime.launch_prime()

  File ~\.conda\envs\myenv\Lib\site-packages\ansys\meshing\prime\internals\launcher.py:241 in launch_prime
    return Client(server_process=server, ip=ip, port=port, timeout=timeout)

  File ~\.conda\envs\myenv\Lib\site-packages\ansys\meshing\prime\internals\client.py:66 in __init__
    self._comm = GRPCCommunicator(

  File ~\.conda\envs\myenv\Lib\site-packages\ansys\meshing\prime\internals\grpc_communicator.py:88 in __init__
    raise ConnectionError(

ConnectionError: Failed to connect to Server in given timeout of 20.0 secs

Please let me know how to resolve it

**Comment by waltersma** [comment, marked as answer]
@ragulselvaraj20 depending on your setup regarding licensing we have seen issues with the timeout not being sufficiently long to allow for a license to be acquired.  Please could you try increasing the timeout to say 60 secs to verify that you can launch the server?

```
import ansys.meshing.prime as prime
prime_client = prime.launch_prime(timeout=60)
```

I hope this helps!  Thanks

**Reply by som1197** [reply]
Hello, it shows me a different error even after adjusting the timeout to 60 seconds.
```

FileNotFoundError                         Traceback (most recent call last)
Cell In[1], line 7
      4 from ansys.meshing import prime
      5 from ansys.meshing.prime.graphics import Graphics
----> 7 prime_client = prime.launch_prime(timeout = 60)
      8 model = prime_client.model
      9 mesh_util = prime.lucid.Mesh(model=model)

File ~\AppData\Local\anaconda3\envs\Ansys_related\Lib\site-packages\ansys\meshing\prime\internals\launcher.py:243, in launch_prime(prime_root, ip, port, timeout, n_procs, version, **kwargs)
    240     client.container_name = container_name
    241     return client
--> 243 server = launch_server_process(
    244     prime_root=prime_root, ip=ip, port=port, n_procs=n_procs, **kwargs
    245 )
    247 return Client(server_process=server, ip=ip, port=port, timeout=timeout)

File ~\AppData\Local\anaconda3\envs\Ansys_related\Lib\site-packages\ansys\meshing\prime\internals\launcher.py:85, in launch_server_process(prime_root, ip, port, n_procs, **kw)
     83     prime_root = get_ansys_prime_server_root()
     84     if prime_root is None:
---> 85         raise FileNotFoundError('No valid Ansys Prime Server found to launch.')
     86 else:  # verify if the file exists
     87     if not os.path.isdir(prime_root):

FileNotFoundError: No valid Ansys Prime Server found to launch.
```

**Reply by waltersma** [reply]
Hi @som1197 , thanks for opening a separate discussion on this #814 as this is a different issue.
